metadata/src/main/resources/common/metadata/PartitionRecord.json

```diff
@@ -47,6 +47,8 @@
       "about": "The eligible leader replicas of this partition." },
     { "name": "LastKnownELR", "type": "[]int32", "default": "null", "entityType": "brokerId",
       "versions": "1+", "nullableVersions": "1+", "taggedVersions": "1+", "tag": 2,
-      "about": "The last known eligible leader replicas of this partition." }
+      "about": "The last known eligible leader replicas of this partition." },
+    { "name": "Directories", "type": "[]uuid", "versions": "2+",
+      "about": "The log directory hosting each replica, sorted in the same exact order as the Replicas field."}
```

Comment:
Not sure, but wondering if this should be a tagged field like the ELR fields.  I don't see any specific advantage to doing so -- we won't ever use this version unless all nodes in the cluster support it, in which case making it mandatory actually makes sense -- but figured I would mention it in case anybody had additional thoughts or context around the ELR choice to tag the new fields in version 1.

Reply:
I would prefer to make it mandatory to make it clear that it's always present in version 2.